lemur/plugins/lemur_telegram/plugin.py

```diff
@@ -0,0 +1,109 @@
+    return ''.join('\\' + c if c in special else c for c in text)
+
+
+def create_certificate_url(name):
+    return "https://{hostname}/#/certificates/{name}".format(
+        hostname=current_app.config.get("LEMUR_HOSTNAME"), name=name
+    )
+
+
+def create_expiration_attachments(certificates):
+    attachments = []
+    for certificate in certificates:
+        name = escape(certificate["name"])
+        owner = escape(certificate["owner"])
+        url = create_certificate_url(certificate["name"])
+        expires = arrow.get(certificate["validityEnd"]).format("dddd, MMMM D, YYYY")
+        endpoints = len(certificate["endpoints"])
+        attachments.append(
+            f"*Certificate:* [{name}]({url})\n*Owner:* {owner}\n*Expires:* {expires}\n*Endpoints:* {endpoints}\n\n"
+        )
+    return attachments
+
+
+def create_rotation_attachments(certificate):
+    name = escape(certificate["name"])
+    owner = escape(certificate["owner"])
+    url = create_certificate_url(certificate["name"])
+    expires = arrow.get(certificate["validityEnd"]).format("dddd, MMMM D, YYYY")
+    endpoints = len(certificate["endpoints"])
+    return f"*Certificate:* [{name}]({url})\n*Owner:* {owner}\n*Expires:* {expires}\n*Endpoints rotated:* {endpoints}\n\n"
+
+
+class TelegramNotificationPlugin(ExpirationNotificationPlugin):
+    title = "Telegram"
+    slug = "tg-notification"
+    description = "Sends certificate expiration notifications to Telegram"
+    version = telegram.VERSION
+
+    author = "Fedor S"
+    author_url = "https://github.com/netflix/lemur"
+
+    additional_options = [
+        {
+            "name": "chat",
+            "type": "str",
+            "required": True,
+            "validation": check_validation("^[+-]?\d+(\.\d+)?$"),
+            "helpMessage": "The chat id to send notification to",
+        },
+        {
+            "name": "token",
+            "type": "str",
+            "required": True,
+            "validation": check_validation("^\d+:[A-Za-z0-9_]+$"),
+            "helpMessage": "Bot API Token",
+        },
+    ]
+
+    def send(self, notification_type, message, targets, options, **kwargs):
+        """
+        A typical check can be performed using the notify command:
+        `lemur notify`
+
+        While we receive a `targets` parameter here, it is unused, plugin currently supports sending to only one chat.
+        """
+        attachments = None
+        if notification_type == "expiration":
+            attachments = create_expiration_attachments(message)
+
+        elif notification_type == "rotation":
+            attachments = create_rotation_attachments(message)
+
+        if not attachments:
+            raise Exception("Unable to create message attachments")
+
+        data = {
+            "parse_mode": "MarkdownV2",
+            "chat_id": self.get_option("chat", options),
+            "text": "*Lemur {} Notification*\n\n{}".format(notification_type.capitalize(), *attachments),
```

Comment:
Looking at the structure, it should probably be:

  # For expiration (list of strings)
  "text": "*Lemur {} Notification*\n\n{}".format(
      notification_type.capitalize(),
      ''.join(attachments)  # Join the list into one string
  )

  # Or for rotation (single string)
  "text": "*Lemur {} Notification*\n\n{}".format(
      notification_type.capitalize(),
      attachments  # Already a string, no unpacking
  )
 as currently unpacking will just yield the first element